src/components/Button/Button.tsx

```diff
@@ -2,33 +2,77 @@ import React from "react";
 import styled, { css } from "styled-components";
 
 export interface ButtonProps {
-  primary?: boolean;
   children: string;
+  variant: "filled" | "outlined";
+  theme: "light" | "dark" | "blue";
+  disabled: boolean;
   onClick?: () => void;
 }
 
 /**
  * Primary UI component for user interaction
  */
 export const Button = ({ children, ...props }: ButtonProps) => (
-  <StyledButton type="button" {...props}>
-    {children}
-  </StyledButton>
+  <StyledButton {...props}>{children}</StyledButton>
 );
 
+const FILLED = ({ theme }: ButtonProps) => css`
+  color: #ffffff;
+  background-color: #0657f9;
+  font-weight: 700;
+
```

Comment:
Maybe add something for color transition. Here is what I usually use
```css
transition-property: color, background-color, border-color;
transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
transition-duration: 150ms;
```
Transition is smoother with that IMO, what do you think ?

Reply:
Good idea, looks much smoother https://github.com/Major-Eth/yearn-web-lib/pull/9/commits/1c5ef95c0c76dae1c16e3b08b4aafdf13cc9c820

Reply:
Removed the `border-color` as that doesn't change onHover https://github.com/Major-Eth/yearn-web-lib/pull/9/commits/69cc60c8c1d5750604054a5dbc6d57151a5ee536